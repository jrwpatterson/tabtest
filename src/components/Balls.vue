<template>
  <div v-bind:class="{ filled: isFilled, powerball: isPowerBall }" class="number-ball">{{ value }}</div>
</template>

<script lang="ts">
import Vue from "vue";
export default Vue.extend({
  props: {
    value: Number,
    isPowerBall: Boolean
  },
  computed: {
    isFilled: function() {
      return this.value != null;
    }
  }
});
</script>

<style scoped lang="scss">
.number-ball {
  height: 44px;
  width: 44px;
  border-radius: 22px;
  margin: 5px;
  border-style: solid;
  border-width: 1px;
  border-color: #d0d0d0;
  display: flex;
  align-items: center;
  justify-content: center;
  box-shadow: 0 -5px 5px -5px #333;

  &.filled {
    color: #ffffff;
    background-color: #1c52a4;
  }

  &.powerball {
    color: #ffffff;
    background-color: #63757f;
  }
}
</style>
