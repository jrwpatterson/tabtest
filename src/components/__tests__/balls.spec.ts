import { render } from "@testing-library/vue";
import Balls from "../Balls.vue";

describe("Balls", () => {
  it("should match its snapshot", () => {
    const { container } = render(Balls, { props: { value: 6 } });
    expect(container).toMatchSnapshot();
  });

  it("should display the correct number", () => {
    const { getByText } = render(Balls, { props: { value: 6 } });
    expect(getByText("6")).toBeDefined();
  });
});
