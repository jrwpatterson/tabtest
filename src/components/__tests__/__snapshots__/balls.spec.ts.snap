// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Balls should match its snapshot 1`] = `
<div>
  <div
    class="number-ball filled"
  >
    6
  </div>
</div>
`;
